client/src/pages/PoolPage.tsx

```diff
@@ -71,18 +78,20 @@ export default function PoolPage() {
     contract.callStatic.get_info()
   );
 
-  const fromMethods = useTokenMethods(coinFrom.assetId);
+  const fromCoinBalance = balances?.find((coin) => coin.assetId === coinFrom.assetId);
   const fromInput = useCoinInput({
     coin: coinFrom,
     coins: getOtherCoins([coinFrom, coinTo]),
     onChangeCoin: (coin: Coin) => setCoins([coin, coinTo]),
+    coinBalance: fromCoinBalance
```

Comment:
You don't need to pass this here, just use the inside functions of the component